src/proto/mod.rs

```diff
@@ -1,28 +1,8 @@
-use failure;
-use std::net::SocketAddr;
-use tokio;
-use tokio::prelude::*;
+//
```

Comment:
Same as for my comment about the `types` module I would move this up one level to a `proto.rs` file.